### PR TITLE
mount: add `--backup-meta` to periodically backup metadata to object storage

### LINKED ISSUE #957: Backup metadata to object store automatically
Right metadata is critical to data safety. If the metadata is lost, we can't get the data back, because we don't know how to assemble the objects in object store back.

People may forget to setup the backup for Redis/MySQL/TiKV, or the backup is broken, we'd like to have another layer to cover the worst case. 

The client (`juicefs mount`) can dump the metadata hourly, and upload them into object store, and keep them based on some rotation policy.  In the worse case, we can download these backup and use it to recover most of the data.

### PR DESCRIPTION
**Pull request by SandyXSD**
close #957

### CODE REVIEW

File: pkg/vfs/backup.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+		time.Sleep(interval / 10)
+		var value []byte
+		if st := m.GetXattr(ctx, 1, key, &value); st != 0 && st != meta.ENOATTR {
+			logger.Warnf("getxattr inode 1 key %s: %s", key, st)
+			continue
+		}
+		var last time.Time
+		var err error
+		if len(value) == 17 { // len("2006-01-02-150405")
+			last, err = time.Parse("2006-01-02-150405", string(value))
```
Comment (davies) [inline pkg/vfs/backup.go:43]: we can use timestamp here

File: pkg/vfs/backup.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+		logger.Warnf("listAll prefix meta/: %s", err)
+		return
+	}
+	var objs []string
+	for o := range objChan {
+		objs = append(objs, o.Key())
+	}
+	sort.Strings(objs)
+
+	// Cleanup policy:
```
Comment (davies) [inline pkg/vfs/backup.go:105]: Add test cases for these

File: pkg/vfs/backup.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+				logger.Infof("backup metadata skipped because of too many inodes: %d %s", iused, interval)
+				continue
+			}
+			val := now.Format("2006-01-02-150405")
+			if st := m.SetXattr(ctx, 1, key, []byte(val), meta.XattrCreateOrReplace); st != 0 {
+				logger.Warnf("setxattr inode 1 key %s: %s", key, st)
+				continue
+			}
+			go cleanupBackups(blob)
+			logger.Infof("backup metadata started")
```
Comment (davies) [inline pkg/vfs/backup.go:62]: debug level

File: pkg/vfs/backup.go [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+			if err = backup(blob, val); err == nil {
+				logger.Info("backup metadata succeed")
```
Comment (davies) [inline pkg/vfs/backup.go:64]: Added the time used by backup.

**Review by davies — CHANGES_REQUESTED** [review]
Unit tests

File: pkg/vfs/backup.go [hunk 5]
```diff
@@ -39,29 +39,28 @@ func Backup(blob object.ObjectStorage, interval time.Duration) {
...
 		if err != nil {
 			logger.Warnf("parse time value %s: %s", value, err)
 			continue
 		}
-		if now := time.Now().UTC(); now.Sub(last) >= interval {
+		if now := time.Now(); now.Sub(last) >= interval {
 			var iused, dummy uint64
 			_ = m.StatFS(ctx, &dummy, &dummy, &iused, &dummy)
-			if iused/(5<<20) > uint64(interval/time.Hour) {
+			if iused/5e6 > uint64(interval/time.Hour) {
```
Comment (davies) [inline pkg/vfs/backup.go:52]: when interval is less than 1 hour, it will not backup anything

File: pkg/vfs/backup.go (hunk 2 above)
Comment (SandyXSD) [reply pkg/vfs/backup.go]: 7e44334

**Review by davies — APPROVED** [review]

**Pull request merged**